root/io/stdpair/pairEnumEvo.C

```diff
@@ -0,0 +1,178 @@
+      return 1;
+   }
+
+   // printInfo("After library loading.");
+
+   auto c = TClass::GetClass(gNameOfMap);
+   // printInfo("After map TClass loading.");
+
+   if (!c)
+      Fatal("pairEnumEvo", "Not TClass for %s", gNameOfMap);
+   c->GetClassInfo();
+   // printInfo("After map TClass GetClassInfo.");
+
+
+   auto p = c->GetCollectionProxy();
+   // printInfo("After CollectionProxy loading.");
+
+   if (fixed)
+      p->GetValueClass();
+ 
+   if (!readbeforeload) {
+      TFile *f1 = nullptr;
+      TFile *f2 = nullptr;
+      if (reportedorder)
+         f1 = TFile::Open("aa02.root");
+      f2 = TFile::Open("ab02.root");
+      if (!reportedorder)
+         f1 = TFile::Open("aa02.root");
+
+      if (!f1)
+         Fatal("pairEnumEvo", "Can not open aa02.root");
+      if (!f2)
+         Fatal("pairEnumEvo", "Can not open ab02.root");
+   }
+
+   // printInfo("After the late file opening.");
+
+   // p->GetValueClass()->GetStreamerInfo()->ls();
+   // TClass::GetClass(gNameOfPairClass)->GetStreamerInfos()->ls();
+
+   fprintf(stdout, "\nLast verifications:\n");
+   fprintf(stdout, "Current StreamerInfo:\n");
+   auto pcl = TClass::GetClass(gNameOfPairClass);
+   auto currentInfo = pcl->GetStreamerInfo();
+   printElements(currentInfo);
+   auto ninfos = pcl->GetStreamerInfos()->GetSize() - 1;
+   for(int i = 1; i < ninfos ; ++i) {
+      auto info = dynamic_cast<TStreamerInfo*>(pcl->GetStreamerInfo(i));
+      if (info && info != currentInfo) {
+         fprintf(stdout, "\n#%d StreamerInfo:\n", i);
+         printElements(info);
+      }
+   }
+   // pcl->GetStreamerInfos()->ls();
+   return 0;
+}
+
+#if 0
+
+Issues with 'interpreted' + load-file-first mode.
+
+Loading the file creates a TClass for the map.  Additional code (eg accessing
+the CollectionProxy) might also create a TClass for the pair.
+
+When the header file is loaded in the interpreter, unless there is explicit
+uses of the map (or pair), there is no decl for the instantiation of thus
+the TClass for the map and pair are not refreshed.
+
+When the pair or map are instantiated (eg. `gInterpreter->Declare("pair<...> pl")`),
+the TClass for the pair is informed (via `TCling::UpdateClassInfoWithDecl`
+and `TCling::RefreshClassInfo`).
+
+We could update `RefreshClassInfo` to refresh the `StreamerInfo` for the pair
+but it would also need to also refresh the map's CollectionProxy (size,
+hints, etc?) [and there is an arbitrary number because they are thread-local]
+
+So at that point, it might actually be better to recreate the TClass for the
+map ...
+
+But wait ... there is currently no support for generating a collection proxy for
+an interpreted class ... so it is actually an emulated collection proxy ...
+
+That proxy does not match the interpreted (nor the compiled) version of the
+map ... so there is no good point to match the pair either ....
+
+So the solution above are (a tad bit) complex and .... not enough ...
```

Comment:
I'm not sure about the conclusion. This is a "won't fix"? Or the test has to wait until the issue is addressed in ROOT?

Reply:
I expanded the note to clarify that the behavior is okay for now but will need to be improved (and thus the test updated) when we support proper CollectionProxy for interpreted STL.